packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -88,11 +89,15 @@ const getLocationSearch = (req) => {
         const values = Array.isArray(value) ? value : [value]
 
         values.forEach((v) => {
-            newParams.append(key, v)
+            // Because URLSearchParams constructor perceives + as space, but append does not
+            // we want to make sure we run the key-value through new URLSearchParams constructor to parse + as space
+            // before we append it to newParams
+            const temp = new URLSearchParams(`${key}=${v}`)
+            newParams.append(key, temp.get(key))
         })
     })
-    const searchString = newParams.toString()
 
+    const searchString = new URLSearchParams(newParams).toString()
```

Comment:
```suggestion
    const searchString = newParams.toString()
```
`newParams` is already a `URLSearchParams`, so you don't need to wrap it.

Reply:
ah my bad, it was not meant to be passed through URLSearchParams again. I reverted to the old code